### PR TITLE
added more base classes for starting test

### PR DESCRIPTION
**Pull request by stadnytskyivp**

### CODE REVIEW

File: src/main/java/com/hotline/elements/Element.java [hunk 1]
```diff
@@ -1,37 +1,89 @@
...
         this.waitBeforeUse = waitBeforeUse;
     }
+
+    public Element(WebElement webElement) {
+        this.by = null;
+        this.waitBeforeUse = false;
+        this.webElement = webElement;
+    }
+
+    protected boolean isDisplayed(int seconds) {
```
Comment (andyyarych) [inline src/main/java/com/hotline/elements/Element.java]: `protected` methods should be declared after `public`, see [Java Code Conventions](https://www.oracle.com/technetwork/java/codeconventions-150003.pdf)

File: src/main/java/com/hotline/elements/Element.java [hunk 2]
```diff
@@ -1,37 +1,89 @@
...
+        } catch (TimeoutException | NoSuchElementException | StaleElementReferenceException ignore) {
+            return false;
+        }
+    }
+
+    public boolean isDisplayed() {
+        return isDisplayed(LONG_WAIT_SECONDS);
+    }
+
+    public WebElement getWebElement() {
```
Comment (andyyarych) [inline src/main/java/com/hotline/elements/Element.java]: I don't really understand what's happening in this method

File: src/main/java/com/hotline/elements/Element.java [hunk 3]
```diff
@@ -1,37 +1,89 @@
...
+import static org.openqa.selenium.support.ui.ExpectedConditions.visibilityOfElementLocated;
 
 public class Element {
-    private final By by;
+    private static final int LONG_WAIT_SECONDS = 20;
+    private static final int SHORT_WAIT_SECONDS = 4;
 
+    private final By by;
     private boolean waitBeforeUse;
     private Element parent;
```
Comment (andyyarych) [inline src/main/java/com/hotline/elements/Element.java]: Where do you define this variable?

File: src/main/java/com/hotline/tests/AbstractTestNG.java [hunk 4]
```diff
@@ -0,0 +1,18 @@
+package com.hotline.tests;
+
+import com.hotline.webdriver.WebDriverFacadeProvider;
+import org.assertj.core.api.AbstractAssert;
+import org.testng.annotations.AfterMethod;
+
+public abstract class AbstractTestNG {
+    private WebDriverFacadeProvider webDriverFacadeProvider;
```
Comment (andyyarych) [inline src/main/java/com/hotline/tests/AbstractTestNG.java]: Where is this variable defined?

File: src/main/java/com/hotline/webdriver/WebDriverFacade.java [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+
+public class WebDriverFacade {
+    private final WebDriver baseDriver;
+
+    public WebDriverFacade(WebDriverFactory webDriverFactory) {
+        this.baseDriver = webDriverFactory.getChromeDriver();
```
Comment (andyyarych) [inline src/main/java/com/hotline/webdriver/WebDriverFacade.java]: You can make `getChromeDriver()` method static and remove input parameter from this constructor

File: src/main/java/com/hotline/webdriver/WebDriverFacade.java [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+    }
+
+    public WebDriver getBaseDriver() {
+        return baseDriver;
```
Comment (andyyarych) [inline src/main/java/com/hotline/webdriver/WebDriverFacade.java]: What if `baseDriver` is not defined? There is a possibility of NullPointerException here

File: src/main/java/com/hotline/webdriver/WebDriverFactory.java [hunk 7]
```diff
@@ -3,13 +3,21 @@
...
+
+import java.util.concurrent.TimeUnit;
 
 public class WebDriverFactory {
     private CurrentEnv currentEnv;
     private MutableCapabilities capabilities;
 
-    WebDriverFactory() {
-        setupWebDriver();
+    public WebDriverFactory(){}
```
Comment (andyyarych) [inline src/main/java/com/hotline/webdriver/WebDriverFactory.java]: Why do you need this line of code?

File: src/main/java/com/hotline/webdriver/WebDriverFactory.java [hunk 8]
```diff
@@ -3,13 +3,21 @@
...
+    public WebDriverFactory(CurrentEnv currentEnv) {
+        this.currentEnv = currentEnv;
+        if ("local".equals(currentEnv.getRunType())){
```
Comment (andyyarych) [inline src/main/java/com/hotline/webdriver/WebDriverFactory.java]: What are other options besides `local`?

File: src/test/java/com/hotline/pages/BaseHotlinePage.java [hunk 9]
```diff
@@ -0,0 +1,11 @@
+package com.hotline.pages;
+
+import com.hotline.modules.HeaderModule;
+
+public class BaseHotlinePage<T extends BaseHotlinePage> {
```
Comment (andyyarych) [inline src/test/java/com/hotline/pages/BaseHotlinePage.java]: Why do you use generics here?

Comment (stadnytskyivp) [reply src/test/java/com/hotline/pages/BaseHotlinePage.java]: @andyyarych You are right! there is no need in them in BaseHotlinePage and HomePage classes.

File: src/main/java/com/hotline/webdriver/WebDriverFactory.java (hunk 7 above)
Comment (stadnytskyivp) [reply src/main/java/com/hotline/webdriver/WebDriverFactory.java]: I have deleted it

File: src/main/java/com/hotline/elements/Element.java (hunk 2 above)
Comment (stadnytskyivp) [reply src/main/java/com/hotline/elements/Element.java]: @andyyarych If you are talking about line 54 I totally agree with you and deleted it

File: src/main/java/com/hotline/webdriver/WebDriverFacade.java (hunk 5 above)
Comment (stadnytskyivp) [reply src/main/java/com/hotline/webdriver/WebDriverFacade.java]: @andyyarych thanks for good advice!

**Review by andyyarych — APPROVED** [review]

**Pull request merged**